### PR TITLE
Newrelic: remove go-cache after compile, Github CI compatible + working

### PR DESCRIPTION
**Pull request by bryanlatten**
- On ARM64 variants, cleans go cache after compilation, saving 37MB
- Collapses variant-specific runtime tests together
- Adds new (basic) Newrelic startup tests
- Full implements GH Actions for CI, doesn't replace Travis CI yet, but is ready

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (bossjones) [inline .github/workflows/ci.yml]: Let's use manual install instead of pipe to sh https://github.com/aelsabbahy/goss#latest

edit: thinking out loud, my concern was for anything that might touch the container and end up in the final artifact. By this stage, the artifact is built and we aren't doing anything like a docker commit after the fact, so this is ok to move forward.

**Review by bossjones — CHANGES_REQUESTED** [review]

**Review by bossjones — APPROVED** [review]

File: test.sh
Comment (adobejmong) [inline test.sh:66]: @bryanlatten : Is there a reason we are using `printf` here? since in lines 57-59 we're using echo (which already has the newline)? Maybe for consistency we should stick to just `echo` unless there's a specific reason we need a different way?

File: scripts/install-newrelic-alpine.sh
Comment (adobejmong) [inline scripts/install-newrelic-alpine.sh:50]: @bryanlatten : More of a question than anything (and its probably small anyway) but I didn't notice `go` being installed in `docker-nginx`. Perhaps not part of this PR but do we keep `go` around or we purge it later on if it's not needed?

File: .github/workflows/ci.yml
Comment (adobejmong) [inline .github/workflows/ci.yml:31]: @bryanlatten : It looks like you already account for it since you're excluding `linux/arm64` in a previous line above  (line 21 I think) but will we need to account for `arm64` if and when it comes? Or should we do the arch string selection in here or a script and since we're excluding arm64, the next time it does become available we only remove it from the exclude?

**Review by adobejmong — APPROVED** [review]
LGTM. Had a few questions but looks pretty good to ship

File: test.sh
Comment (bryanlatten) [reply test.sh:66]: it was only consistency, we could switch one or switch them all. your call

File: scripts/install-newrelic-alpine.sh
Comment (bryanlatten) [reply scripts/install-newrelic-alpine.sh:50]: @adobejmong `go` isn't installed in docker-nginx? Surrounding this script installs and removes it. Note: the first version of this script had all in-script dependencies installed/removed together, but on removal it would also wipe out dependencies that were still needed...so we're tracking them all at the top level.

File: .github/workflows/ci.yml
Comment (bryanlatten) [reply .github/workflows/ci.yml:31]: @adobejmong yes, it gets super complicated at that point. Note: these instances of goss are *outside* the container, meaning they evaluate the assembled package. `dgoss` specifically pokes a hole in the container and mounts its own `goss` for testing. So, we will have to account for an x64 machine emulating a arm64 container run and poking/testing on that. Definitely for a followup (I welcome contribution!)

File: test.sh
Comment (adobejmong) [reply test.sh:66]: Thanks @bryanlatten . Yes it would be worthwhile to have them be consistent but we can deal with it some other time. Like I mentioned not enough to block the PR.

File: scripts/install-newrelic-alpine.sh
Comment (adobejmong) [reply scripts/install-newrelic-alpine.sh:50]: Perfect. Thanks for clarifying.

**Pull request merged**
